spec/slack-ruby-bot/commands/commands_before_and_after_command_spec.rb

```diff
@@ -0,0 +1,347 @@
+require 'spec_helper'
+
+describe SlackRubyBot::Commands do
+  def app
+    SlackRubyBot::App.new
+  end
+
+  before(:all) do
+    @klass = Class.new(SlackRubyBot::Commands::Base) do
```

Comment:
Well, that hasn't gone well. `ActiveSupport::Callbacks` really only works on instance methods of a class. The entire `command` structure is setup as class methods so there is no object with instance methods to use for the callbacks.

See here for an explanation:  http://stackoverflow.com/questions/12870244/activesupportcallbacks-on-class-methods

I have another idea that I'm working on where `command` work is split up into the model-view-controller pattern. Right now `command` acts as the controller with the model & view logic embedded in it. I think this can be improved for better testability, so I'm going to hack on that for a while. That approach, which will use explicit classes, might lend itself better to using `ActiveSupport::Callbacks`. I'll send a PR.

Let's leave this one open for a bit...